### PR TITLE
Fix `Scrollbar` thumb drag behavior on desktop.

### LINKED ISSUE #107765: ScrollBar moves with an offset if cursor leaves flutter application.
This issue is reproducible on both the stable and master branch, and affects both desktop and web applications. I checked the behaviour with web apps such as YouTube and the chrome browser and it seems that they take into consideration the position of the cursor when deciding as to whether to move the ScrollBar.

<details>

<summary>video of bug in Flutter:</summary>

https://user-images.githubusercontent.com/73116038/179329223-c0bf6738-3cfa-4f8f-bcdc-6a88b5e9e6bb.mp4

</details>

<details>

<summary>video of expected behaviour in YouTube:</summary>

https://user-images.githubusercontent.com/73116038/179329228-da40bd35-8cb1-43fb-80b3-fe756663769d.mp4

</details>

<details>

<summary>Reproducible code:</summary>

    import 'package:flutter/material.dart';
    
    void main() => runApp(const MyApp());
    
    class MyApp extends StatelessWidget {
      const MyApp({Key? key}) : super(key: key);
    
      static const String _title = 'Flutter Code Sample';
    
      @override
      Widget build(BuildContext context) {
        return MaterialApp(
          title: _title,
          home: Scaffold(
            appBar: AppBar(title: const Text(_title)),
            body: const Center(
              child: MyStatefulWidget(),
            ),
          ),
        );
      }
    }
    
    class MyStatefulWidget extends StatefulWidget {
      const MyStatefulWidget({Key? key}) : super(key: key);
    
      @override
      State<MyStatefulWidget> createState() => _MyStatefulWidgetState();
    }
    
    class _MyStatefulWidgetState extends State<MyStatefulWidget> {
      final ScrollController _firstController = ScrollController();
    
      @override
      Widget build(BuildContext context) {
        return Scrollbar(
          thumbVisibility: true,
          controller: _firstController,
          child: ListView.builder(
              controller: _firstController,
              itemCount: 100,
              itemBuilder: (BuildContext context, int index) {
                return Padding(
                  padding: const EdgeInsets.all(8.0),
                  child: Text('Scrollable 1 : Index $index'),
                );
              }),
        );
      }
    }

</details>

<details>

<summary>flutter doctor -v</summary>

    [✓] Flutter (Channel master, 3.1.0-0.0.pre.1633, on LMDE 5 (elsie)
        5.10.0-16-amd64, locale en_GB.UTF-8)
        • Flutter version 3.1.0-0.0.pre.1633 on channel master at
          /home/henryr/Dev/SDK/Flutter/flutter
        • Upstream repository https://github.com/flutter/flutter.git
        • Framework revision 95a5a8d741 (2 days ago), 2022-07-13 09:53:40 -0700
        • Engine revision 20038216ac
        • Dart version 2.18.0 (build 2.18.0-272.0.dev)
        • DevTools version 2.15.0
    
    [✓] Android toolchain - develop for Android devices (Android SDK version
        32.1.0-rc1)
        • Android SDK at /home/henryr/Android/Sdk
        • Platform android-32, build-tools 32.1.0-rc1
        • Java binary at:
          /home/henryr/Dev/SDK/AndroidStudio/android-studio/jre/bin/java
        • Java version OpenJDK Runtime Environment (build
          11.0.12+0-b1504.28-7817840)
        • All Android licenses accepted.
    
    [✓] Chrome - develop for the web
        • Chrome at google-chrome
    
    [✓] Linux toolchain - develop for Linux desktop
        • Debian clang version 11.0.1-2
        • cmake version 3.18.4
        • ninja version 1.10.1
        • pkg-config version 0.29.2
    
    [✓] Android Studio (version 2021.2)
        • Android Studio at /home/henryr/Dev/SDK/AndroidStudio/android-studio
        • Flutter plugin version 69.0.2
        • Dart plugin version 212.5744
        • Java version OpenJDK Runtime Environment (build
          11.0.12+0-b1504.28-7817840)
    
    [✓] Connected device (2 available)
        • Linux (desktop) • linux  • linux-x64      • LMDE 5 (elsie) 5.10.0-16-amd64
        • Chrome (web)    • chrome • web-javascript • Google Chrome 103.0.5060.114
    
    [✓] HTTP Host Availability
        • All required HTTP hosts are available
    
    • No issues found!

</details>

### PR DESCRIPTION
**Pull request by TahaTesser**
fixes https://github.com/flutter/flutter/issues/107765

`macOS`, `Windows`, and `Linux` support dragging scrollbar from only where the thumb is pressed.

Demo: https://github.com/flutter/flutter/pull/108166#discussion_r928108576

<details> 
<summary>Code Sample</summary> 

```dart
import 'package:flutter/material.dart';

void main() => runApp(const MyApp());

class MyApp extends StatelessWidget {
  const MyApp({super.key, this.dark = true, this.useMaterial3 = true});

  final bool dark;
  final bool useMaterial3;

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      debugShowCheckedModeBanner: false,
      themeMode: dark ? ThemeMode.dark : ThemeMode.light,
      theme: ThemeData(
        useMaterial3: useMaterial3,
        brightness: Brightness.light,
        colorSchemeSeed: const Color(0xff6750a4),
      ),
      darkTheme: ThemeData(
        useMaterial3: useMaterial3,
        brightness: Brightness.dark,
        colorSchemeSeed: const Color(0xff6750a4),
      ),
      home: const Example(),
    );
  }
}

class Example extends StatelessWidget {
  const Example({super.key});

  @override
  Widget build(BuildContext context) {
    final ScrollController scrollController = ScrollController();
    final UniqueKey uniqueKey = UniqueKey();
    return Directionality(
      textDirection: TextDirection.ltr,
      child: MediaQuery(
        data: const MediaQueryData(),
        child: ScrollConfiguration(
          behavior: const ScrollBehavior().copyWith(
            scrollbars: false,
          ),
          child: PrimaryScrollController(
            controller: scrollController,
            child: RawScrollbar(
              isAlwaysShown: true,
              thumbColor: const Color(0xffff0000),
              controller: scrollController,
              child: CustomScrollView(
                primary: true,
                // scrollDirection: Axis.horizontal,
                // center: uniqueKey,
                slivers: <Widget>[
                  SliverToBoxAdapter(
                    child: Container(
                      height: 600.0,
                    ),
                  ),
                  SliverToBoxAdapter(
                    key: uniqueKey,
                    child: Container(
                      height: 600.0,
                    ),
                  ),
                  SliverToBoxAdapter(
                    child: Container(
                      height: 600.0,
                    ),
                  ),
                ],
              ),
            ),
          ),
        ),
      ),
    );
  }
}
``` 
	
</details>

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by TahaTesser** [comment]
cc: @werainkhatri 
Since you're the one who pointed out this behavior, do you wanna take a look at this PR?

**Comment by werainkhatri** [comment]
> cc: @werainkhatri 
> Since you're the one who pointed out this behavior, do you wanna take a look at this PR?

hey! ya it's in my to-do. am a little busy this week. will try to pick it up if a get a bit of time. hope that's okay.

File: packages/flutter/lib/src/widgets/scrollbar.dart [hunk 1]
```diff
@@ -1785,6 +1787,8 @@ class RawScrollbarState<T extends RawScrollbar> extends State<T> with TickerProv
     _fadeoutTimer?.cancel();
     _fadeoutAnimationController.forward();
     _dragScrollbarAxisOffset = localPosition;
+    _thumbPressVertical = localPosition.dy - scrollbarPainter._thumbOffset;
+    _thumbPressHorizontal = localPosition.dx - scrollbarPainter._thumbOffset;
```
Comment (Piinks) [inline packages/flutter/lib/src/widgets/scrollbar.dart:1791]: Do we need to keep track of both the horizontal and the vertical? Or just the axis we care about? It looks like you filter for it later in `handleThumbPressUpdate`

**Review by Piinks — COMMENTED** [review]
I checked this out today and played around with it. Very nice, elegant.

File: packages/flutter/lib/src/widgets/scrollbar.dart (hunk 1 above)
Comment (TahaTesser) [reply packages/flutter/lib/src/widgets/scrollbar.dart]: Great idea, I cleaned this up. 

But for some reason, the golden file change message is thrown. This shouldn't have any golden changes AFAIK.

**Review by Piinks — APPROVED** [review]
This LGTM! Thanks!

The golden file change is not related to your PR. That test has been flaky and should be skipped now if you update your branch. We are tracking that in a separate issue as well - sorry for the inconvenience!

**Pull request merged**